app/parsers/katz.py

```diff
@@ -0,0 +1,124 @@
+from plugins.stockpile.app.parsers.base_parser import BaseParser
+from plugins.stockpile.app.relationship import Relationship
+from app.utility.logger import Logger
+
+import re
+from collections import defaultdict
+
+class MimikatzBlock(object):
+    def __init__(self):
+        self.session = ''
+        self.username = ''
+        self.domain = ''
+        self.logon_server = ''
+        self.logon_time = ''
+        self.sid = ''
+        self.packages = defaultdict(list)
+
+
+class Parser(BaseParser):
+    def __init__(self, parser_info):
+        self.mappers = parser_info['mappers']
+        self.used_facts = parser_info['used_facts']
+        self.parse_mode = 'wdigest'
+        self.log = Logger('parsing_svc')
+
+    def parse_kats(self, output):
+        """
+        Parses mimikatz output with the logonpasswords command and returns a list of dicts of the credentials.
+        Args:
+            output: stdout of "mimikatz.exe privilege::debug sekurlsa::logonpasswords exit"
+        Returns:
+            A list of MimikatzSection objects
+        """
+        # split sections using "Authentication Id" as separator
+        sections = output.split('Authentication Id')
+        creds = []
+        for section in sections:
+            mk_section = MimikatzBlock()
+            package = {}
+            package_name = ''
+            in_header = True
+            for line in section.splitlines():
+                line = line.strip()
+                if in_header:
+                    if line.startswith('msv'):
+                        in_header = False
+                    else:
+                        session = re.match(r'^\s*Session\s*:\s*([^\r\n]*)', line)
+                        if session:
+                            mk_section.session = session.group(1)
+                        username = re.match(r'^\s*User Name\s*:\s*([^\r\n]*)', line)
+                        if username:
+                            mk_section.username = username.group(1)
+                        domain = re.match(r'^\s*Domain\s*:\s*([^\r\n]*)', line)
+                        if domain:
+                            mk_section.domain = domain.group(1)
+                        logon_server = re.match(r'^\s*Logon Server\s*:\s*([^\r\n]*)', line)
+                        if logon_server:
+                            mk_section.logon_server = logon_server.group(1)
+                        logon_time = re.match(r'^\s*Logon Time\s*:\s*([^\r\n]*)', line)
+                        if logon_time:
+                            mk_section.logon_time = logon_time.group(1)
+                        sid = re.match(r'^\s*SID\s*:\s*([^\r\n]*)', line)
+                        if sid:
+                            mk_section.sid = sid.group(1)
```

Comment:
Can we make this a private function? This public function is very large

Reply:
I think I found a way.